pkg/adapter/mtping/runner.go

```diff
@@ -99,6 +100,9 @@ func (a *cronJobsRunner) AddSchedule(cfg PingConfig) cron.EntryID {
 		ResourceGroup: resourceGroup,
 	}
 	ctx = kncloudevents.ContextWithMetricTag(ctx, metricTag)
+	ctx = logging.ContextWithMTLoggerFromLogger(ctx, a.Logger, cfg.Namespace)
```

Comment:
The call to `ctx := context.Background()` in mtping `AddSchedule` will not respect the sigterm signals from the main thread as it is currently written.